pydantic/main.py

```diff
@@ -418,9 +433,25 @@ def __iter__(self) -> 'AnyGenerator':
         """
         yield from self._iter()
 
-    def _iter(self, by_alias: bool = False) -> 'TupleGenerator':
+    def _iter(self, by_alias: bool = False, skip_defaults: bool = True) -> 'TupleGenerator':
         for k, v in self.__values__.items():
-            yield k, self._get_value(v, by_alias=by_alias)
+            yield k, self._get_value(v, by_alias=by_alias, skip_defaults=skip_defaults)
+
+    def _calculate_keys(
```

Comment:
I think we can probably do this in a more efficient way, eg. if `include=None`, `exclude=set()` and `skip_defaults=False` (all the defaults), then we just take all keys.

Can't we take care of that case but just returning `None` or something here and not checking key containment?

Reply:
Sure, I can work on optimizing that out.

Reply:
On this note, the include/exclude defaults are not consistent (copy for example has them both defaulting to None). I can clean this up as well.

Reply:
ok.